File: examples/access/locations/polygon_from_vertices.py
# /// script
# dependencies = ["brahe"]
# ///
"""
Create a PolygonLocation from a list of vertices.
Demonstrates polygon construction and center/vertex access.
"""

import brahe as bh

bh.initialize_eop()

# Define polygon vertices (longitude, latitude, altitude)
# Simple rectangular region
vertices = [
    [-122.5, 37.7, 0.0],
    [-122.35, 37.7, 0.0],
    [-122.35, 37.8, 0.0],
    [-122.5, 37.8, 0.0],
    [-122.5, 37.7, 0.0],  # Close the polygon
]

polygon = bh.PolygonLocation(vertices).with_name("SF Region")

print(f"Name: {polygon.get_name()}")
print(f"Vertices: {polygon.num_vertices}")
print(
    f"Center: ({polygon.longitude(bh.AngleFormat.DEGREES):.4f}, {polygon.latitude(bh.AngleFormat.DEGREES):.4f})"
)

# Expected output:
# Name: SF Region
# Vertices: 4
# Center: (-122.4250, 37.7500)

File: examples/access/locations/polygon_from_vertices.rs
//! Create a PolygonLocation from a list of vertices.
//! Demonstrates polygon construction and center/vertex access.

#[allow(unused_imports)]
use brahe as bh;
use bh::utils::Identifiable;
use bh::AccessibleLocation;
use nalgebra as na;

fn main() {
    bh::initialize_eop().unwrap();

    // Define polygon vertices (lon, lat, alt in degrees and meters)
    let vertices = vec![
        na::SVector::<f64, 3>::new(-122.5, 37.7, 0.0),
        na::SVector::<f64, 3>::new(-122.35, 37.7, 0.0),
        na::SVector::<f64, 3>::new(-122.35, 37.8, 0.0),
        na::SVector::<f64, 3>::new(-122.5, 37.8, 0.0),
        na::SVector::<f64, 3>::new(-122.5, 37.7, 0.0),
    ];

    let polygon = bh::PolygonLocation::new(vertices).unwrap()
        .with_name("SF Region");

    let center = polygon.center_geodetic();
    println!("Name: {}", polygon.get_name().unwrap_or_default());
    println!("Vertices: {}", polygon.num_vertices());
    println!("Center: ({:.4}, {:.4})", center[0], center[1]);

    // Expected output:
    // Name: SF Region
    // Vertices: 4
    // Center: (-122.4250, 37.7500)
}
